### PR TITLE
New module - meraki_firewalled_services

### LINKED ISSUE #48613: New module - meraki_firewalled_services
##### SUMMARY
Meraki added support for firewalled services in the API so Meraki modules should support this.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
meraki_network

### PR DESCRIPTION
**Pull request by kbreit**
##### SUMMARY
- Module can query or modify network services on Meraki gear
- Check mode supported
- Integration tests included

Fixes #48613

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
meraki_firewalled_services

##### ADDITIONAL INFORMATION
```paste below
    data:
    - access: blocked
      service: ICMP
    - access: restricted
      allowedIps:
      - 192.0.1.1
      - 192.0.1.2
      service: web
```

### CODE REVIEW

File: lib/ansible/modules/network/meraki/meraki_firewalled_services.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    meraki = MerakiModule(module, function='firewalled_services')
+    module.params['follow_redirects'] = 'all'
+
+    net_services_urls = {'firewalled_services': '/networks/{net_id}/firewalledServices'}
+    services_urls = {'firewalled_services': '/networks/{net_id}/firewalledServices/{service}'}
+
+    meraki.url_catalog['network_services'] = net_services_urls
+    meraki.url_catalog['service'] = services_urls
+
+    if meraki.params['net_name'] and meraki.params['net_id']:
```
Comment (ganeshrn) [inline lib/ansible/modules/network/meraki/meraki_firewalled_services.py]: Can this be added while creating an object of AnsibleModule?
Something similar to 
```
mutually_exclusive = [('net_name',  'net_id')]
module = AnsibleModule(argument_spec=argument_spec,
                           supports_check_mode=True,
                           mutually_exclusive=mutually_exclusive
                           )

```

**Review by Qalthos — COMMENTED** [review]
I agree with Ganesh's note, otherwise it looks fine.

File: lib/ansible/modules/network/meraki/meraki_firewalled_services.py (hunk 1 above)
Comment (kbreit) [reply lib/ansible/modules/network/meraki/meraki_firewalled_services.py]: Fixed. Thank you.

**Comment by kbreit** [comment]
@ganeshrn or @Qalthos - Can you please review and merge if happy?

**Review by Qalthos — APPROVED** [review]

**Comment by Qalthos** [comment]
shipit

**Review by pabelanger — APPROVED** [review]

**Comment by kbreit** [comment]
@pabelanger With two approvals shouldn’t this auto merge?

**Comment by kbreit** [comment]
@pabelanger I realized I didn't have integration tests here. I added the file.

**Pull request merged**